C++ Interop: Add support for `char`

bricknerb opened the pull request:
Added tests for different character types.
`char` is currently not in primitives prelude, so had to use full prelude.

C++ Interop Demo:

```carbon
// main.carbon

library "Main";

import Cpp inline '''
auto output_char(char c) -> void {
  printf("%c", c);
}
''';

fn Run() -> i32 {
  let msg: array(Core.Char, 13) =
      ('H', 'e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd', '!', '\n');
  for (c: Core.Char in msg) {
    Cpp.output_char(c);
  }
  return 0;
}
```

```shell
$ bazel-bin/toolchain/carbon compile main.carbon
$ bazel-bin/toolchain/carbon link main.o --output=demo
$ ./demo
Hello world!
```

Part of https://github.com/carbon-language/carbon-lang/issues/5263.

zygoloid commented on toolchain/check/import_cpp.cpp:
```diff
@@ -1080,6 +1080,11 @@ static auto MapBuiltinType(Context& context, SemIR::LocId loc_id,
                       context.types().GetInstId(GetSingletonType(
                           context, SemIR::BoolType::TypeInstId)));
   }
+  if (type.isCharType() && ast_context.getTypeSize(qual_type) == 8) {
```
This will incorrectly match `signed char` and `unsigned char` as well as plain `char`. You could either use `isSpecificBuiltinType(Char_U) || isSpecificBuiltinType(Char_S)`, or use `ast_context.hasSameType(qual_type, ast_context.CharTy)` to match only plain `char` and not the other two.

bricknerb replied:
Thanks, done.

bricknerb committed:
C++ interop: Add support for `char`
Added tests for different character types.
`char` is currently not in primitives prelude, so had to use full prelude.

C++ Interop Demo:

```carbon
// main.carbon

library "Main";

import Cpp inline '''
auto output_char(char c) -> void {
  printf("%c", c);
}
''';

fn Run() -> i32 {
  let msg: array(Core.Char, 13) =
      ('H', 'e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd', '!', '\n');
  for (c: Core.Char in msg) {
    Cpp.output_char(c);
  }
  return 0;
}
```

```shell
$ bazel-bin/toolchain/carbon compile main.carbon
$ bazel-bin/toolchain/carbon link main.o --output=demo
$ ./demo
Hello world!
```

Part of #5263.

bricknerb committed:
Use `TypeExpr::None`.
